### ISSUE TITLE
"Common parts" option has no effect

### ISSUE THREAD

**Issue by gyim** [gyim/breviar-ios#6]
For celebrations of saints, e.g. St Thomas Aquinas, they sometimes have more options to take "common texts". Thomas Aq: (1) for priests, (2) for Church doctors.
The global setting does not apply -- app always chooses the first option from specific saint.
Maybe this is problem of CGI app (not iOS app).

**Comment by kominak** [comment]
@breviar-sk could you please provide precise URL example here? I'll compare it with what app generates and I'll try to fix it somehow.

**Comment by gyim** [comment]
You can see an example on the following page:

http://zsolozsma.katolikus.hu/cgi-bin/l.cgi?qt=pdt&d=28&m=1&r=2015&p=mpc&ds=1&j=hu&o1=1439&o2=56&o3=4

Click on "közös rész kiválasztása" (výber spoločnej časti...) and choose "lelkipásztorok - papok" from the second select box ("a közös részből" / "časti modlitby zo spoločnej časti"). Click on "Ima megjelenítése" (Zobraz modlitbu) on the bottom of the page => the text of the first hymn should change (only in the Hungarian version, for some reason).

I tried to set the same option in the iOS app, but it always loaded the same hymn.

**Comment by kominak** [comment]
Fixed

**Comment by kominak** [comment]
@breviar-sk:
> After option o3 is changed (it works! Thank you Oto for that fix.) in Settings, app must re-read the current date. (Test scenario: Select Jan 31, choose in SK version: for salesian family (SDB), then return to 31 Jan – there should be in uppercase long celebration name for Don Bosco. When you change the day (prev, next), everything is OK. Now the same vice versa: select general calendar (for SK). Jan 31 should contain only ordinary memory of Don Bosco, not solemnity – again, when you change the day (prev, next), everything is OK.)

**Comment by gyim** [comment]
The bug above is fixed in commit e547850. It was not caused by the of3 parameter, the reason was that the calendar ("k") parameter was not used passed generating the list of celebrations. And the reason why it worked at all was that the "k" parameter was passed later to the CGI, when we preloaded the prayers. The CGI module stored this parameter somewhere and used it the next time we generated the list of celebrations :)
